src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -95,4 +100,12 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         const config = await getClientConfig(this.parent, postgresDefaultDatabase);
         await runPostgresQuery(config, `Drop Database ${wrapArgInQuotes(this.databaseName)};`);
     }
+
+    public async isFirewallRuleSet(treeItem: PostgresServerTreeItem): Promise<boolean> {
```

Comment:
This should probably have a comment in the code explaining why we're doing it. Something like:
> Flexible servers throw a generic 'ETIMEDOUT' error instead of the firewall-specific error, so we have to check the firewall rules